Products/ZenModel/actions.py

```diff
@@ -392,77 +393,111 @@ def _encodeBody(self, body):
         else:            
             plain_body = MIMEText(body.decode('ascii', 'ignore'))
         return plain_body
-    
+
+    def _targetsByTz(self, dmd, targets):
+        """
+        Take timezone from user property to convert a event time in
+        notification and also group targets emails by those timezones.
+        """
+        targets_by_timezones = {}
+        for user in dmd.ZenUsers.getAllUserSettings():
+            if user.email in targets:
+                if targets_by_timezones.get(user.timezone):
+                    targets_by_timezones[user.timezone].add(user.email)
+                else:
+                    targets_by_timezones[user.timezone] = set([user.email])
```

Comment:
Also, `targets_by_timezones` is quite a longish name, I would stick with `tz_targets` or something like that